en/editor/preferences/index.md

```diff
@@ -0,0 +1,81 @@
+
+- **WeChatGame APP Path**: Set the WechatGame App Path, See [WeChat Mini Games](../publish/publish-wechatgame.md).
+
+- **Android NDK Path**: Set the NDK Path, See [Setup Native Development Environment](../publish/setup-native-development.md).
+
+- **Android SDK Path**: Set Android SDK Path, See [Setup Native Development Environment](../publish/setup-native-development.md).
+
+## Data Editor
+
+This category is used to set the default open mode for resources.
+
+![edit](./index/edit.jpg)
+
+- **Script Editor**：You can use the executable file of any external text editing tool as the opening method when you double-click the script file in `Assets`.
+
+- **Picture Editor**: Similar to the above option, used here to set when `Assets`, double-click the picture file, open the picture with the default application path.
+
+## Extension
+
+The plugin allows to directly register configuration items into preferences.The configuration items registered in the preferences will be displayed here.For details, please refer to the documentation of each plug-in.
+
+## Laboratory
+
+About Laboratory：
+
+1. The laboratory provides some new technical solutions or experimental features, you can turn each feature on with the on/off switch.
+
+2. Non-backward compatible changes or removal may occur in any future release. Use of the feature in production environments should be well tested and pay attention to the publish notes of future versions.
+
+3. We welcome users to try out and give us your feed backs via our [forum](https://discuss.cocos2d-x.org/), you can make those features more powerful and easier to use in your own projects.
+
+### Use The Search Pop To Add Components
+
+Many users feedback that components can not be searched when add components make they inconvenient. In the face of massive custom scripts component of many projects, find their own scripting components is very difficult.But use the search pop means multi-step operation. Of course, you can turn this option back to the original use.
+
+![add components](./index/add-component.jpg)
+
+In order to make user more convenient, we also add some auxiliary editing design:
+
+- The focus will be in the search box when the pop-up window is opened, and you can directly search for related keywords without mouse click to expand;
+
+- When searching for related components, you can directly use the shortcut keys `↑ ↓` to quickly switch component options, and press the `enter` key to add the component;
+
+For this feature, if you have more awesome suggestions, welcome to tell us in the [forum](https://discuss.cocos2d-x.org/).
+
+### Turn on terrain
+
+More details can refer to [Terrain system](../terrain/index.md).
```

Comment:
More details can refer to [Terrain system](../terrain/index.md).
-->
For more details please refer to the [Terrain system](../terrain/index.md).